specification/app/resource-manager/Microsoft.App/preview/2024-02-02-preview/JavaComponents.json

```diff
@@ -401,62 +401,141 @@
-              "x-ms-enum": {
-                "name": "JavaComponentProvisioningState",
-                "modelAsString": true
-              }
-            },
-            "configurations": {
-              "description": "List of Java Components configuration properties",
-              "type": "array",
-              "items": {
-                "$ref": "#/definitions/JavaComponentConfigurationProperty"
-              },
-              "x-ms-identifiers": [
-                "propertyName"
-              ]
-            },
-            "serviceBinds": {
-              "description": "List of Java Components that are bound to the Java component",
-              "type": "array",
-              "items": {
-                "$ref": "#/definitions/JavaComponentServiceBind"
-              },
-              "x-ms-identifiers": [
-                "name"
-              ]
-            }
+          "$ref": "#/definitions/JavaComponentProperties"
+        }
+      }
+    },
+    "JavaComponentProperties": {
+      "description": "Java Component common properties.",
+      "type": "object",
+      "required": [
+        "componentType"
+      ],
+      "properties": {
+        "componentType": {
+          "description": "Type of the Java Component.",
+          "enum": [
+            "SpringBootAdmin",
+            "SpringCloudEureka",
+            "SpringCloudConfig",
+            "Nacos"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "JavaComponentType",
+            "modelAsString": true
+          }
+        },
+        "provisioningState": {
+          "description": "Provisioning state of the Java Component.",
+          "enum": [
+            "Succeeded",
+            "Failed",
+            "Canceled",
+            "Deleting",
+            "InProgress"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "JavaComponentProvisioningState",
+            "modelAsString": true
+          }
+        },
+        "configurations": {
+          "description": "List of Java Components configuration properties",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/JavaComponentConfigurationProperty"
+          },
+          "x-ms-identifiers": [
+            "propertyName"
+          ]
+        },
+        "serviceBinds": {
+          "description": "List of Java Components that are bound to the Java component",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/JavaComponentServiceBind"
           },
-          "x-ms-client-flatten": true
+          "x-ms-identifiers": [
+            "name"
+          ]
+        }
+      },
+      "x-ms-client-flatten": true,
+      "discriminator": "componentType"
+    },
+    "JavaComponentIngress": {
+      "description": "Container App Ingress configuration.",
+      "type": "object",
+      "properties": {
+        "fqdn": {
```

Comment:
Would it better to be `dashboardFqdn` since it will enforce AAD auth and block the APP/SDK calls?